pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -34,54 +33,88 @@ func (d *DualWriterMode2) Create(ctx context.Context, obj runtime.Object, create
 		return created, err
 	}
 
-	c, err := enrichObject(obj, created)
+	accessorCreat, err := meta.Accessor(created)
 	if err != nil {
 		return created, err
 	}
 
-	accessor, err := meta.Accessor(c)
+	accessorOld, err := meta.Accessor(obj)
 	if err != nil {
 		return created, err
 	}
 
+	enrichObject(accessorOld, accessorCreat)
+
 	// create method expects an empty resource version
-	accessor.SetResourceVersion("")
-	accessor.SetUID("")
+	accessorCreat.SetResourceVersion("")
+	accessorCreat.SetUID("")
 
-	rsp, err := d.Storage.Create(ctx, c, createValidation, options)
+	rsp, err := d.Storage.Create(ctx, created, createValidation, options)
 	if err != nil {
 		klog.FromContext(ctx).Error(err, "unable to create object in Storage", "mode", 2)
 	}
 	return rsp, err
 }
 
-func enrichObject(orig, copy runtime.Object) (runtime.Object, error) {
-	accessorC, err := meta.Accessor(copy)
-	if err != nil {
-		return nil, err
-	}
-	accessorO, err := meta.Accessor(orig)
-	if err != nil {
-		return nil, err
-	}
-
+func enrichObject(accessorO, accessorC metav1.Object) {
 	accessorC.SetLabels(accessorO.GetLabels())
 
 	ac := accessorC.GetAnnotations()
 	for k, v := range accessorO.GetAnnotations() {
 		ac[k] = v
 	}
 	accessorC.SetAnnotations(ac)
-
-	// #TODO set resource version and UID when required (Update for example)
-	// accessorC.SetResourceVersion(accessorO.GetResourceVersion())
-
-	// accessorC.SetUID(accessorO.GetUID())
-
-	return copy, nil
 }
 
 // Get overrides the default behavior of the Storage and retrieves an object from LegacyStorage
 func (d *DualWriterMode2) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
 	return d.Legacy.Get(ctx, name, &metav1.GetOptions{})
 }
+
+// Update overrides the generic behavior of the Storage and writes first to the legacy storage and then to US.
```

Comment:
Might be helpful to refer to Storage rather than Unified Storage to match the naming of the Storage struct. Also, using [this test](https://github.com/grafana/grafana/blob/main/pkg/tests/apis/playlist/playlist_test.go#L87-L98) as the best example I could think of, the DW is used for a variety of "storage" types. This includes "unified" as well as "file".

```suggestion
// Update overrides the behavior of the generic DualWriter and writes first to LegacyStorage and then to Storage.
```